### ISSUE TITLE
Schema restrictions on BigQueryExampleGen

### ISSUE THREAD

**Issue by casassg** [tensorflow/tfx#2709]
We are getting an error when trying to load wide datasets with BigQueryExampleGen.
```
DataflowRuntimeException: Dataflow pipeline failed. State: FAILED, Error:
Workflow failed. Causes: S22:InputToRecord[train]/QueryTable/ReadFromBigQuerySource+InputToRecord[train]/ToTFExample+WriteSplit[train]/MaybeSerialize+WriteSplit[train]/Shuffle/AddRandomKeys+WriteSplit[train]/Shuffle/ReshufflePerKey/Map(reify_timestamps)+WriteSplit[train]/Shuffle/ReshufflePerKey/GroupByKey/Reify+WriteSplit[train]/Shuffle/ReshufflePerKey/GroupByKey/Write failed., Table schema exceeds limits. Size is 1067276 bytes.  Limit is 1048576 bytes.
```

As per internal reports, it seems this restriction may be fixed with `ReadFromBigQuery` (and not in `Read(BigQuerySource)`).  Seems Apache Beam 2.25 (release a few days ago) sets the default to `ReadFromBigQuery`. Is the idea to set the default to `ReadFromBigQuery` in next release?

**Comment by rmothukuru** [comment]
Please provide details about what platform you are using (operating system, architecture). Also include your TFX version. 

Make sure you also include the exact command if possible to produce the output included in your test case. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by casassg** [comment]
May be worth adding the template to this repo.

- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Dataflow worker. Scheduled from CentOS. 
- TensorFlow installed from (source or binary): custom binary 
- TensorFlow version (use command below): 1.15
- Python version: 3.7.9
- TFX version: 0.24.1
- Apache Beam version: 2.24.0

**Comment by rmothukuru** [comment]
@casassg,
Sure, we are in the process of adding that template. Thank you for your feedback.

In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by casassg** [comment]
I unfortunatly can't copy the code snipped due to being an internal production database. Will try to reproduce error with open datasets.

**Comment by zhitaoli** [comment]
> 
> As per internal reports, it seems this restriction may be fixed with `ReadFromBigQuery` (and not in `Read(BigQuerySource)`). Seems Apache Beam 2.25 (release a few days ago) sets the default to `ReadFromBigQuery`. Is the idea to set the default to `ReadFromBigQuery` in next release?

Yes we plan to do this, possibly in the next 1-2 releases.

We have received more performance benchmark results from apache-beam team who claimed that the new `ReadFromBigQuery` is almost performance parity now, so we plan to do to switch after beam 2.25. This probably will result in the actual change landing in our Nov release, to give us a bit time to integrate and test.

With that said, are you able to build and run TFX from either a branch, or use our [nightly builds](https://www.tensorflow.org/tfx/guide#nightly_packages)? If so, you might be able to test this out before we officially release it.

**Comment by casassg** [comment]
We created a fork for now of the BigQueryExampleGen that forces usage of `ReadFromBigQuery` on apache beam 2.24 (less performance but we were looking to get unlocked in short term). 

I can run TFX from master from my local tfx setup, I can try to run once the upgrade lands on master to check that it's fixed.

**Comment by zhitaoli** [comment]
TFX 0.25.0 has been released and switched to use `ReadFromBigQuery` by default. Can you let us know if this or your local fork fixes the issue, @casassg ?

I would proceed to close this issue since there is not known action item TFX could take. Feel free to reopen if the issue persists after these changes.

**Comment by casassg** [comment]
@zhitaoli we plan to land the upgrade next week aprox. Will let you know if we still see an issue 😄
